src/vs/workbench/contrib/debug/browser/watchExpressionsView.ts

```diff
@@ -214,19 +222,34 @@ export class WatchExpressionsView extends ViewPane {
 
 	private onContextMenu(e: ITreeContextMenuEvent<IExpression>): void {
 		const element = e.element;
+		const context = this.getWatchContext(element);
 		const selection = this.tree.getSelection();
 
 		this.watchItemType.set(element instanceof Expression ? 'expression' : element instanceof Variable ? 'variable' : undefined);
 		const actions: IAction[] = [];
 		const attributes = element instanceof Variable ? element.presentationHint?.attributes : undefined;
 		this.variableReadonly.set(!!attributes && attributes.indexOf('readOnly') >= 0 || !!element?.presentationHint?.lazy);
-		createAndFillInContextMenuActions(this.menu, { arg: element, shouldForwardArgs: true }, actions);
+		createAndFillInContextMenuActions(this.menu, { shouldForwardArgs: true }, actions);
 		this.contextMenuService.showContextMenu({
 			getAnchor: () => e.anchor,
 			getActions: () => actions,
-			getActionsContext: () => element && selection.includes(element) ? selection : element ? [element] : [],
+			getActionsContext: (_e, contributedAction) => contributedAction ? context : element && selection.includes(element) ? selection : element ? [element] : [],
 		});
 	}
+
+	private getWatchContext(expression: IExpression | null): IWatchContext | undefined {
+		if (!expression) {
+			return undefined;
+		}
+
+		return {
+			sessionId: this.debugService.getViewModel().focusedSession?.getId(),
+			expressionId: expression.getId(),
```

Comment:
I don't think this ID is meaningful or exposed to extensions elsewhere, or is it?

Reply:
You could be right, but it may be useful as a unique ID?